source/extensions/common/wasm/wasm_vm.cc

```diff
@@ -72,11 +72,27 @@ bool EnvoyWasmVmIntegration::getNullVmFunction(std::string_view function_name, b
   return false;
 }
 
+absl::string_view getWasmEngine() {
+  absl::string_view wasm_ext_runtime;
+  for (const auto& ext : Envoy::Registry::FactoryCategoryRegistry::registeredFactories()) {
+    if (ext.first == "envoy.wasm.runtime") {
+      for (auto engine_name : ext.second->registeredNames()) {
+        if (engine_name != "envoy.wasm.runtime.null") {
+          wasm_ext_runtime = engine_name;
+        }
+      }
+    }
+  }
+  return wasm_ext_runtime;
+}
+
 WasmVmPtr createWasmVm(absl::string_view runtime) {
+  // Set wasm runtime to build-in wasm engine if it is not specified
   if (runtime.empty()) {
-    ENVOY_LOG_TO_LOGGER(Envoy::Logger::Registry::getLog(Envoy::Logger::Id::wasm), warn,
-                        "Failed to create Wasm VM with unspecified runtime");
-    return nullptr;
+    ENVOY_LOG_TO_LOGGER(Envoy::Logger::Registry::getLog(Envoy::Logger::Id::wasm), info,
+                        "Wasm VM runtime is unspecified, set to envoy built-in wasm runtime: {}",
+                        getWasmEngine());
```

Comment:
We're changing this field to optional with a clearly defined default value, so I'd completely remove this log entry.

Reply:
Log entry is removed.